document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java

```diff
@@ -102,4 +114,25 @@ private Collector getCollectorForLogicalOperator(LogicalOperator operator) {
     throw new UnsupportedOperationException(
         String.format("Query operation:%s not supported", operator));
   }
+
+  private boolean isOperatorNeedsFieldAccessor(RelationalOperator operator) {
+    switch (operator) {
+      case CONTAINS:
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  private Object convertStringValueToMap(Object value) {
+    if (value instanceof String) {
```

Comment:
Can we do this a bit cleanly?
Maybe, ConstantExpression can take in a document and handle the case.
Please refer to the way it is done in `NestedSubDocumentValue`.
Basically, we should be able to differentiate a literal String vs a JSON String.